### PR TITLE
feature/backend invitation processing

### PR DESCRIPTION
**Pull request by JGiter**
- `IOrganizationInvitation` extended with `sender` property to display invitation sender to the invited user 
- add `Viewed` status to `IOrganizationInvitationStatus` enum to track whether invitation was communicated
- `organization` property type of `IOrganizationInvitation` changed to `IOrganization`
- Changes to `OrganizationInvitation` entity are migrated
- Since invitation updating now is not only about acceptance or rejection `OrganizationInvitationService.acceptOrReject` renamed to `update` and its logic is changed

### CODE REVIEW

File: packages/origin-backend-core/src/OrganizationInvitation.ts [hunk 1]
```diff
@@ -20,11 +20,9 @@ export interface IOrganizationInvitationProperties {
 }
 
 export interface IOrganizationInvitation extends IOrganizationInvitationProperties {
-    organization: IOrganization | IOrganization['id'];
-}
-
-export interface IOrganizationInvitationWithRelations extends IOrganizationInvitation {
     organization: IOrganization;
+    sender: string;
+    processed: boolean;
```
Comment (kosecki123) [inline packages/origin-backend-core/src/OrganizationInvitation.ts]: Can we use the status of the invitation to represent if invitation was received or not?

So IMHO adding extra statuses to already existing enum should solve the problem:

OrganizationInvitationStatus.Accepted
OrganizationInvitationStatus.Rejected
OrganizationInvitationStatus.Sent
OrganizationInvitationStatus.Delivered

Is there a use case which requires us to have more than 1 status at time?

File: packages/origin-backend/migrations/1596616659110-OrganizationInvitationSenderEmail.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class OrganizationInvitationSenderAndProcessed1596616659110 implements MigrationInterface {
+    name = 'OrganizationInvitationSenderAndProcessed1596616659110';
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        await queryRunner.query(
+            `ALTER TABLE "organization_invitation" ADD "sender" character varying NOT NULL`,
```
Comment (kosecki123) [inline packages/origin-backend/migrations/1596616659110-OrganizationInvitationSenderEmail.ts]: Adding NOT NULL columns require a default value or will not migrate already existing DB records and fail the process.

**Review by kosecki123 — CHANGES_REQUESTED** [review]
Please check my comments.

File: packages/origin-backend-core/src/OrganizationInvitation.ts [hunk 3]
```diff
@@ -2,9 +2,10 @@ import { IOrganization } from './Organization';
 import { Role } from './User';
 
 export enum OrganizationInvitationStatus {
-    Pending,
     Rejected,
-    Accepted
+    Accepted,
+    Sent,
+    Viewed
```
Comment (kosecki123) [inline packages/origin-backend-core/src/OrganizationInvitation.ts:8]: Do not reorder those enums, as existing records in DB will be incorrectly mapped:

Previous Pending = Rejected
Rejected = Accepted ....

So either append only or assign int numbers explicitly so they are not messed up.

**Review by kosecki123 — CHANGES_REQUESTED** [review]
Enum reorder.

File: packages/origin-backend-core/src/OrganizationInvitation.ts [hunk 4]
```diff
@@ -2,9 +2,10 @@ import { IOrganization } from './Organization';
 import { Role } from './User';
 
 export enum OrganizationInvitationStatus {
-    Pending,
-    Rejected,
-    Accepted
+    Rejected = 1,
+    Accepted = 2,
+    Sent = 3,
```
Comment (kosecki123) [inline packages/origin-backend-core/src/OrganizationInvitation.ts]: What's the different between Sent and Pending? What happen to existing Pending invitations?

File: packages/exchange/src/pods/order/order.controller.ts [hunk 5]
```diff
@@ -84,7 +84,7 @@ export class OrderController {
     }
 
     @Get()
-    @UseGuards(AuthGuard(), ActiveUserGuard)
+    @UseGuards(AuthGuard())
```
Comment (kosecki123) [inline packages/exchange/src/pods/order/order.controller.ts]: Why?

Comment (JGiter) [reply packages/exchange/src/pods/order/order.controller.ts]: According to https://energyweb.atlassian.net/wiki/spaces/OR/pages/1012989957/Organizations+and+users all users have access to Exchange where orders are listed

Comment (kosecki123) [reply packages/exchange/src/pods/order/order.controller.ts]: /orders endpoint is for managing organization orders. Non active user should not be able to use it.

**Review by kosecki123 — APPROVED** [review]

**Pull request merged**